gnomad_qc/v4/annotations/vrs_annotation_batch_v4.py

```diff
@@ -0,0 +1,523 @@
+            vcf_input = f"gs://{working_bucket}/vrs-temp/shards/shard-{version}.vcf.bgz/{vcf_name}"
+            vcf_output = f'/temp-vcf-annotated/annotated-{vcf_name.split(".")[0]}.vcf'
+
+            # Perform VRS annotation on vcf_input and store output in /vrs-temp
+            new_job.command("mkdir /temp-vcf-annotated/")
+            new_job.command(
+                f"python3 {vrs_script_path} --vcf_in"
+                f" {batch_vrs.read_input(vcf_input)} --vcf_out"
+                f" {vcf_output} --seqrepo_root_dir {seqrepo_path} --assembly"
+                f" {assembly} --vrs_attributes"
+            )
+
+            # bgzip the annotated VCF
+            new_job.command(f"bgzip {vcf_output}")
+
+            # Copy annotated shard to its appropriate place in Google Bucket
+            new_job.command(
+                "gsutil cp"
+                f" {vcf_output}.gz"
+                f" gs://{working_bucket}/vrs-temp/annotated-shards/annotated-{version}.vcf/"
+            )
+
+        # Execute all jobs in the batch_vrs Batch
+        batch_vrs.run()
+        logger.info(
+            "Batch jobs executed, annotated vcf.gz shards copied to"
+            f" gs://{working_bucket}/vrs-temp/annotated-shards/annotated-{version}.vcf"
+        )
+
+    if args.annotate_original:
+        logger.info(
+            "Batch jobs executed, preparing to read in sharded VCF from prior step."
+            " Preparing list of files first using Hail's hadoop_ls method."
+        )
+
+        annotated_file_dict = hl.utils.hadoop_ls(
+            f"gs://{working_bucket}/vrs-temp/annotated-shards/annotated-{version}.vcf/"
+        )
+        # NOTE: this is changed to be a directory intstead `/*.vcf` because the
+        # QoS step was run with Hail 0.2.115 instead of the latest version
+
+        annotated_file_list = [
+            annotated_file_item["path"] for annotated_file_item in annotated_file_dict
+        ]
+
+        logger.info("File list created. Now reading in annotated shards.")
+
+        # Import all annotated shards
+        ht_annotated = hl.import_vcf(
+            annotated_file_list,
+            force_bgz=True,
+            reference_genome=assembly,
+        ).make_table()
+
+        logger.info("Annotated table constructed")
+
+        vrs_struct = hl.struct(
+            VRS_Allele_IDs=ht_annotated.info.VRS_Allele_IDs.split(","),
+            VRS_Starts=ht_annotated.info.VRS_Starts.split(",").map(lambda x: hl.int(x)),
+            VRS_Ends=ht_annotated.info.VRS_Ends.split(",").map(lambda x: hl.int(x)),
+            VRS_States=ht_annotated.info.VRS_States.split(","),
+        )
+
+        ht_annotated = ht_annotated.annotate(vrs=vrs_struct)
+        ht_annotated = ht_annotated.select(ht_annotated.vrs)
+
+        logger.info("Adding VRS IDs and GA4GH.VRS version to original Table")
+        ht_final = ht_original.annotate(
+            vrs=ht_annotated[ht_original.locus, ht_original.alleles].vrs
```

Comment:
does the input ht not have an info field?

Reply:
No, it doesn't have an info field yet, because I wrote VDS out to a simple HT, I will add  `info` to the `create_release_HT` when I merge all the annotations.